test/functional/api/extmark_spec.lua

```diff
@@ -1423,6 +1423,48 @@ describe('API/extmarks', function()
     eq({ {1, 0, 0}, {2, 0, 8} },
         meths.buf_get_extmarks(0, ns, 0, -1, {}))
   end)
+
+  it("works with undofile", function()
+    os.remove("testfile")
+    os.remove(".testfile.un~")
+    command("e! testfile")
+    command("set undodir=. | set undofile")
+
+    insert(
+    [[12345
+hello world]])
+
+    eq({"12345", "hello world"}, curbufmeths.get_lines(0, -1, true))
+
+    curbufmeths.set_extmark(ns, 0, 0, {})
+
+    -- splice
+    feed("gg0d2l")
+
+    eq({"345", "hello world"}, curbufmeths.get_lines(0, -1, true))
+
+    -- move
+    command(".m+1")
+
+    eq({"hello world", "345"}, curbufmeths.get_lines(0, -1, true))
+
+    -- reload undofile and undo changes
+    command("w")
+    command("set noundofile")
+    command("bw!")
+    command("e! testfile")
+
+    command("set undofile")
+
+    feed("u")
+    eq({"345", "hello world"}, curbufmeths.get_lines(0, -1, true))
```

Comment:
hmm, all of this behavior still works from before right? perhaps it is easier to test with `lua/buffer_update_spec.lua` in the `on_bytes` part (getting accurate byte-level updates).

Reply:
Makes sense. I've updated the tests